### PR TITLE
Add updateValueTimeout to editor (T589214)

### PR DESCRIPTION
**Pull request by sometimesElen**
Add updateValueTimeout to editor, provide this option to search mixin (TreeView and List search) and get rid of extra timeout in Grids

### CODE REVIEW

File: js/ui/shared/ui.editor_factory_mixin.js [hunk 1]
```diff
@@ -39,33 +39,35 @@ var EditorFactoryMixin = (function() {
 
     var getTextEditorConfig = function(options) {
         var isValueChanged = false,
-            data = {},
-            isEnterBug = checkEnterBug(),
-            sharedData = options.sharedData || data;
+            isEnterBug = checkEnterBug();
```
Comment (DokaRus) [inline js/ui/shared/ui.editor_factory_mixin.js]: Rename it, pls. It's not clear what do you check. I'll need to view the whole file to know that you check a current device and browser.

File: js/ui/editor/editor.js [hunk 2]
```diff
@@ -303,15 +331,16 @@ var Editor = Widget.inherit({
...
-                    this.validationRequest.fire({
-                        value: args.value,
-                        editor: this
-                    });
+                var argument;
+                window.clearTimeout(that._valueChangeTimeout);
+
+                if(args.value !== "" && that.option("updateValueTimeout")) {
+                    argument = args;
+                    that._valueChangeTimeout = window.setTimeout(function() {
```
Comment (DokaRus) [inline js/ui/editor/editor.js]: So you **always** wrap `_valueOptionChange` into timeout for a base editor? I don't think that it's a good idea for common scenarios.

**Review by DokaRus — CHANGES_REQUESTED** [review]
I think it's a too dangerous PR for 17.2. It may lead to potential breaking changes in the next minor update.

**Comment by DokaRus** [comment]
I think @mrjohn0011 and @lukyanovas should check this PR too

**Review by vconst — CHANGES_REQUESTED** [review]
I think that updateValueTimeout logic should be moved to another place. The updateValueTimeout should work with value bindings also. So value option should be changed on timeout.

File: js/ui/editor/editor.js (hunk 2 above)
Comment (sometimesElen) [reply js/ui/editor/editor.js]: No, I Wrap it only if timeout is set.

**Comment by lukyanovas** [comment]
I think that this logic should be separated from texteditor

**Review by mrjohn0011 — CHANGES_REQUESTED** [review]
I think text editor should not have such possibility. The text editor does not know about a search or custom server requests that a user performs on value change. The user should set a timeout in a handler. This pull request should be closed

**Comment by sometimesElen** [comment]
After further consideration our team decided to implement a timeout on a search mixin level.

**Pull request closed without merging**